### PR TITLE
Swift uses `containter` not `bucket`

### PR DESCRIPTION
**Pull request by rullzer**
Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

### CODE REVIEW

**Review by MorrisJobke — APPROVED** [review]

**Review by icewind1991 — CHANGES_REQUESTED** [review]
While the swift library *does* use `container`, the connection logic will properly use `bucket` when available and the multi bucket code works with `bucket`

So in the config using `bucket` would be prefered, for the `cacheKey` `container` is good.

**Comment by rullzer** [comment]
@icewind1991 in that case https://github.com/nextcloud/server/blob/817f2f8f49aec409da832219f43663793e1933a8/lib/private/Files/ObjectStore/Swift.php#L65 also need to be properly fixed... I don't care what we use but this mixing and matching has to be cleaned up

**Comment by icewind1991** [comment]
The setup code sets `container` to `bucket` so "inside" the swift code `container` will always work while using `bucket` will break for setups that have configured `container` in their config

**Comment by rullzer** [comment]
Ok then let me fix it all in one go... because currently we also get a 'non defined offset' with the code

**Comment by rullzer** [comment]
@icewind1991 please have another look this should do the trick then...

File: lib/private/Files/ObjectStore/SwiftFactory.php [hunk 1]
```diff
@@ -77,7 +77,7 @@ private function getClient() {
 			$this->params['container'] = $this->params['bucket'];
 		}
 		if (!isset($this->params['container'])) {
-			$this->params['container'] = 'owncloud';
```
Comment (icewind1991) [inline lib/private/Files/ObjectStore/SwiftFactory.php]: there is a small chance that changing this will break an existing installation

**Review by icewind1991 — APPROVED** [review]

**Pull request merged**